src/ansys/fluent/core/fluent_connection.py

```diff
@@ -285,6 +361,95 @@ def id(self) -> str:
+                f"Cannot execute cleanup script, Fluent running inside container. "
+                f"Use kill_fluent_container() instead."
+            )
+            return
+        if self._remote_instance is not None:
+            logger.error(f"Cannot execute cleanup script, Fluent running remotely.")
+            return
+
+        pwd = self.connection_properties.cortex_pwd
+        pid = self.connection_properties.cortex_pid
+        host = self.connection_properties.cortex_host
+        cleanup_line_start = "Cleanup script file is "
+        if os.name == "nt":
+            cleanup_line_end = ".bat\n"
+            cmd_list = []
+        elif os.name == "posix":
+            cleanup_line_end = ".sh\n"
+            cmd_list = ["bash"]
+        else:
+            logger.error(
+                f"Unrecognized or unsupported operating system, cancelling Fluent cleanup script execution."
+            )
+            return
+        list_dir = os.listdir(Path(pwd))
+        for name in list_dir:
+            if name.startswith("fluent-") and name.endswith(f"{pid}.trn"):
+                with open(name, "r") as file:
+                    trn_lines = file.readlines()
+                for line in trn_lines:
+                    if (
+                        line.startswith(cleanup_line_start)
+                        and line.endswith(cleanup_line_end)
+                        and host in line
+                    ):
+                        cleanup_filename = line.lstrip(cleanup_line_start).rstrip()
+                        cleanup_filepath = Path(pwd, cleanup_filename)
+                        logger.info(
+                            f"Executing Fluent cleanup script, filepath: {cleanup_filepath}"
+                        )
+                        cmd_list.append(cleanup_filepath)
+                        logger.debug(f"Cleanup command list = {cmd_list}")
+                        subprocess.Popen(
+                            cmd_list,
+                            stdout=subprocess.DEVNULL,
+                            stderr=subprocess.DEVNULL,
+                        )
+
+    def kill_fluent_container(self):
+        """
+        Immediately kills the docker container running the Fluent client,
+        losing unsaved progress and data.
+
+        Notes
+        -----
+        By default, Fluent does not run in a container,
+        in that case use :func:`kill_fluent()`.
+        If the Fluent session is responsive, prefer using :func:`exit()` instead.
+        """
+        if not self.connection_properties.inside_container:
+            logger.error(
+                f"Session is not inside a container, cannot kill Fluent container. "
+                f"Use kill_fluent() instead."
+            )
+            return
+        if self._remote_instance is not None:
+            logger.error(f"Fluent is running remotely, cannot kill Fluent container.")
+            return
+        container_id = self.connection_properties.cortex_host
```

Comment:
Where are we storing the container-id?

Reply:
We are not storing the container id directly, only the cortex_host variable inside `connection_properties`. The hostname of the container is the same as the container id, which can be used for docker commands.